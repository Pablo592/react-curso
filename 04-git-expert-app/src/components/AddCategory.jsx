import React, { useState } from 'react'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setINputValue] = useState('Dragon Ball')

    const onINputChange = ({target}) => {
        setINputValue([target.value])
    }

    const onSubmit = (event) => {
        event.preventDefault()

         if( inputValue.toString().trim().length < 1) return 
            setCategories(categories => [inputValue, ...categories])
            setINputValue('')
    }

    return (
        <>
            <h2>Agregue una nueva categoria</h2>
            <form onSubmit={onSubmit}>
                <input 
                    type="text"
                    name='cate' 
                    placeholder="Buscar gifts" 
                    value={ inputValue } 
                    onChange={onINputChange} 
                />
            </form>
        </>
    )
}
