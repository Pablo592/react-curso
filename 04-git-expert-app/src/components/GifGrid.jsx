import {useState, useEffect} from 'react'
import { getGifs } from '../helpers/getGifs'




export const GifGrid = ({ category }) => {


    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs(category)

    }, [])

    getGifs(category)


  return (

    <>
        <h3>{ category }</h3>
    </>
    
  )
}
