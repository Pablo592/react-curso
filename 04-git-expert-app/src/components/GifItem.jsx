import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({title,images,id}) => {
  return (
    <div className='card'>
        <img src={images.downsized.url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  images: PropTypes.object.isRequired,
}