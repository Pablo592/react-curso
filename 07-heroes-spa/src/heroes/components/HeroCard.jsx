import { Link } from 'react-router-dom';

const charactersByHero = (alter_ego, characters) => {
    if (alter_ego !== characters) {
        return <p>{characters}</p> ;
    } else {
        return '';
    }
}

export const HeroCard = ({
    id,
    superhero,
    alter_ego,
    first_appearance,
    characters,
    publisher
}) => {

    const imageUrl = `../../assets/heroes/${id}.jpg`

    return (
        <div className='card ms-3' style={{ maxWidth: 540 }}>
            <div className='row no-gutters'>
                <div className='col-md-4'>
                    <img src={imageUrl} className='card-img' alt={superhero} />
                </div>
                <div className='col-md-8'>
                    <div className='card-body'>
                        <h5 className='card-title'>{superhero}</h5>
                        <p className='card-text'>{alter_ego}</p>
        
                        {
                            charactersByHero(alter_ego, characters)
                        }
                        <p className='card-text'>
                            <small className='text-muted'>{first_appearance}</small>
                        </p>
                        <Link to={`/hero/${id}`}>
                            More...
                        </Link>
                    </div>
                </div>
            </div>
        </div>
    )
}
