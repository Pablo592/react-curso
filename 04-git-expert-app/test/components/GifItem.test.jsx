import React from 'react';
import { render } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem.jsx>', () => {
  
    const title = 'Saitama'
    const images = {downsized: {url: 'https://localhost/saitama.jpg'}}
    const id = '1234'

    test('debe de hacer match con el snapshot ', () => {
      

        const {container} = render( <GifItem title={title} images={images} id={id} /> )

        expect(container).toMatchSnapshot()
    })
    
})
