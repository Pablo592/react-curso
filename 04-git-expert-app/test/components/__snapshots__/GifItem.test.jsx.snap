// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx> debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://localhost/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
